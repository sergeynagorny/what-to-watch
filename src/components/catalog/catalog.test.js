import React from "react";
import renderer from "react-test-renderer";
import Catalog from "./catalog.jsx";


const films = [
  {
    id: 1,
    title: `What We Do in the Shadows`,
    poster: `img/what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 2,
    title: `Pulp Fiction`,
    poster: `img/pulp-fiction.jpg`,
  }
];

it(`Should Catalog render correctly`, () => {
  const tree = renderer.create(
      <Catalog
        films={films}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
