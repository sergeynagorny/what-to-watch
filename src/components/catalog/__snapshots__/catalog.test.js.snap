// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Catalog render correctly 1`] = `
<div
  className="page-content"
>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <ul
      className="catalog__genres-list"
    >
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          All genres
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Comedies
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Crime
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Documentary
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Dramas
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Horror
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Kids &amp; Family
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Romance
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Sci-Fi
        </a>
      </li>
      <li
        className="catalog__genres-item"
      >
        <a
          className="catalog__genres-link"
          href="#"
          onClick={[Function]}
        >
          Thrillers
        </a>
      </li>
    </ul>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            poster="img/what-we-do-in-the-shadows.jpg"
            width="280"
          >
            <source />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            poster="img/pulp-fiction.jpg"
            width="280"
          >
            <source />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="movie-page.html"
          >
            Pulp Fiction
          </a>
        </h3>
      </article>
    </div>
    <div
      className="catalog__more"
    >
      <button
        className="catalog__button"
        type="button"
      >
        Show more
      </button>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
